source/opt/instruction.cpp

```diff
@@ -409,9 +414,18 @@ bool Instruction::IsVulkanUniformBuffer() const {
   return is_block;
 }
 
-bool Instruction::IsReadOnlyVariableShaders() const {
-  uint32_t storage_class = GetSingleWordInOperand(kVariableStorageClassIndex);
+bool Instruction::IsReadOnlyPointerShaders() const {
+  if (!type_id()) {
```

Comment:
type_id is not a bool, so I prefer a comparison with 0.

```
if (type_id == 0) {
```

Reply:
Done.